server/routes/auth.js

```diff
@@ -0,0 +1,93 @@
+const express = require("express");
+const jwt = require("jsonwebtoken");
+
+const userController = require("../controllers/usersController");
+
+const router = express.Router();
+
+router.post("/login", validationMiddleware, async function (req, res, next) {
+    try {
+        const { email, password } = req.body;
+        // verify that there is user for email
+        const user = await userController.findOneWithemail(email);
+        if (!user) {
+            res.status(400).json({ errors: ["Email or password invalid"] });
+            return;
+        }
+        // verify that the passwords match
+        const passwordsMatch = await userController.checkPassword({
+            user,
+            password,
+        });
+
+        if (!passwordsMatch) {
+            res.status(400).json({ errors: ["Email or password invalid"] });
+            return;
+        }
+        // create and return jwt with user obj
+        const responseObj = await createResponseObj(user._doc);
+        res.cookie("token", responseObj.token, { httpOnly: true });
+        res.json(responseObj);
+    } catch (error) {
+        console.error(error);
+        next(error);
```

Comment:
We should probably just return a 400 here, right?

Reply:
As we aren't sure what went wrong for the code to reach this catch, so I think we should return 500 as it is likely something that went wrong on the server side.